packages/Search/src/details/DTK_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -106,26 +106,79 @@ struct DistributedSearchTreeImpl
+          View::rank == 1 && !std::is_same<typename View::traits::array_layout,
+                                           Kokkos::LayoutStride>::value ) &&
+        std::is_same<typename View::traits::memory_space,
+                     typename View::traits::host_mirror_space::memory_space>::
+            value )>::type * = 0 )
+{
+    return Kokkos::View<typename View::traits::non_const_data_type,
+                        Kokkos::LayoutRight,
+                        typename View::traits::host_mirror_space>(
+        std::string( src.label() ).append( "_layout_right_mirror" ),
+        src.dimension_0(), src.dimension_1(), src.dimension_2(),
+        src.dimension_3(), src.dimension_4(), src.dimension_5(),
+        src.dimension_6(), src.dimension_7() );
+}
+
+template <typename View>
+inline Kokkos::View<typename View::traits::non_const_data_type,
+                    Kokkos::LayoutRight,
+                    typename View::traits::host_mirror_space>
+create_layout_right_mirror_view(
+    View const &src,
+    typename std::enable_if<(
+        ( std::is_same<typename View::traits::array_layout,
+                       Kokkos::LayoutRight>::value ||
+          View::rank == 1 && !std::is_same<typename View::traits::array_layout,
+                                           Kokkos::LayoutStride>::value ) &&
+        std::is_same<typename View::traits::memory_space,
+                     typename View::traits::host_mirror_space::memory_space>::
+            value )>::type * = 0 )
+{
+    return src;
+}
+
 template <typename DeviceType>
-template <typename T>
-void DistributedSearchTreeImpl<DeviceType>::sendAcrossNetwork(
-    Tpetra::Distributor &distributor, Kokkos::View<T *, DeviceType> exports,
-    Kokkos::View<T *, DeviceType> imports )
+template <typename View>
+typename std::enable_if<Kokkos::is_view<View>::value>::type
+DistributedSearchTreeImpl<DeviceType>::sendAcrossNetwork(
+    Tpetra::Distributor &distributor, View exports,
+    typename View::non_const_type imports )
 {
-    // NOTE: this function encapsulates the communication from and to views
-    // with data living on the device.  This is a workaround we will
-    // (hopefully) get rid of in the future when we upgrade Trilinos.  We
-    // should be able to directly call doPostsAndWaits() on the views passed as
-    // argument.  See https://github.com/trilinos/Trilinos/issues/1454
-    auto exports_host = Kokkos::create_mirror_view( exports );
+    DTK_REQUIRE( ( exports.dimension_1() == imports.dimension_1() ) &&
+                 ( exports.dimension_2() == imports.dimension_2() ) &&
+                 ( exports.dimension_3() == imports.dimension_3() ) &&
+                 ( exports.dimension_4() == imports.dimension_4() ) &&
+                 ( exports.dimension_5() == imports.dimension_5() ) &&
+                 ( exports.dimension_6() == imports.dimension_6() ) &&
+                 ( exports.dimension_7() == imports.dimension_7() ) );
```

Comment:
Make sure to trigger exception in test.  Also might want to check that distributor and view sizes are compatible.

Reply:
Added check for compatible distributor and exports/imports views.

Reply:
Didn't bother adding a test to trigger though.